crates/sequencer/src/sequencer.rs

```diff
@@ -931,36 +930,57 @@ where
 
         // TODO: create register_healthcheck_rpc helper function
         let ledger_db = self.ledger_db.clone();
-        rpc.register_method("health_check", move |_, _| {
-            let next_soft_batch_num = ledger_db.get_next_items_numbers().soft_batch_number;
-            if next_soft_batch_num < 2 {
-                return Ok::<(), ErrorObjectOwned>(());
-            }
-
-            let soft_batches = ledger_db
-                .get_soft_batch_range(
-                    &(BatchNumber(next_soft_batch_num - 2)..BatchNumber(next_soft_batch_num)),
-                )
-                .map_err(|err| {
+        rpc.register_async_method("health_check", move |_, _| {
+            let ledger_db = ledger_db.clone();
+            async move {
+                let head_batch = ledger_db.get_head_soft_batch().map_err(|err| {
```

Comment:
why not make the whole closure async move?

Reply:
It caused all move and lifetime issues. I tried Arc and everything but only cloning `ledger_db` before every invocation of the async part of the code solved the issue.

Reply:
@kpp @rakanalh any better way to do this?

if not, @yaziciahmet make sure clone on ledger db is a simple operation

Reply:
we can also use the RpcContext to share the ledger_db. The reason I didn't go that way initially is, full node is not using RpcContext but sequencer is using it, hence it was easier to clone the ledger_db and use it. I will try to make RpcContext shared in common crate and use it in both places.

Reply:
that makes more sense, however, the sequencer has mempool in its context and the prover does not. what about that?

Reply:
Might be a bit dirty, but we can have a trait in common called `LedgerDBAccessor`, and the actual context implementations can be different for Sequencer and FullNode. And these contexts would implement this trait simply returning the LedgerDB.

Reply:
maybe.

your call :)

Reply:
I believe it is unnecessary to deal with contexts right now as cloning LedgerDB seems to be really cheap due to its' `db` and `next_item_numbers` fields being encapsulated with an Arc, and as far as I researched, it is also a cheap operation to clone a broadcast channel. Hence I will keep the implementation as it is.